src/backend/distributed/utils/citus_clauses.c

```diff
@@ -29,6 +29,32 @@ static Expr * citus_evaluate_expr(Expr *expr, Oid result_type, int32 result_typm
 
 
 /*
+ * Whether the executor needs to reparse and try to execute this query.
+ */
+bool
+RequiresMasterEvaluation(Query *query)
+{
+	ListCell *targetEntryCell = NULL;
+
+	foreach(targetEntryCell, query->targetList)
```

Comment:
I think it's ok to leave this for 5.2.  But i'd be surprised if it weren't measurable, when running many concurrent operations on the master ( a common workload for citus).